pkg/ccl/backupccl/restore_job.go

```diff
@@ -1509,7 +1515,7 @@ func (r *restoreResumer) doResume(ctx context.Context, execCtx interface{}) erro
 		if err := r.cleanupTempSystemTables(ctx, nil /* txn */); err != nil {
 			return err
 		}
-	} else if details.RestoreSystemUsers {
+	} else if details.DescriptorCoverage == tree.SystemUsers {
```

Comment:
It makes me a little uneasy thinking about what copy of details we're reading from here, and if we have indeed set and persisted `tree.SystemUsers` in a details that is coming from an older node. How do you feel about being more defensive and writing a helper method `isSystemUserRestore` which will check `details.DescriptorCoverage == tree.SystemUsers || details.RestoreSystemUsers`. We can them remove this helper and only check coverage in 23.1?

Reply:
very good call. Done.